app/models/spree/user.rb

```diff
@@ -35,9 +35,9 @@ class User < ApplicationRecord
     has_many :enterprise_roles, dependent: :destroy
     has_many :enterprises, through: :enterprise_roles
     has_many :owned_enterprises, class_name: 'Enterprise',
-                                 foreign_key: :owner_id, inverse_of: :owner
+                                 foreign_key: :owner_id, inverse_of: :owner, dependent: :nullify
     has_many :owned_groups, class_name: 'EnterpriseGroup',
-                            foreign_key: :owner_id, inverse_of: :owner
+                            foreign_key: :owner_id, inverse_of: :owner, dependent: :nullify
```

Comment:
Enterprises and groups must have an owner. I think that we should raise when trying to delete here. That's probably already happening because the foreign key constraint is violated. Nullifying would allow us to delete these users and have orphaned, invalid enterprises in the database.

Reply:
changing to :restrict_with_exception